core/src/main/java/org/opensearch/sql/calcite/utils/WildcardUtils.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.calcite.utils;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/** Utility class for wildcard pattern matching in field names. */
+public class WildcardUtils {
+
+  private static final String WILDCARD_CHAR = "*";
+
+  public static boolean matchesWildcardPattern(String pattern, String fieldName) {
+
+    if (!pattern.contains(WILDCARD_CHAR)) {
+      return pattern.equals(fieldName);
+    }
+
+    String[] parts = pattern.split("\\*", -1);
+    String regex =
+        String.join(
+            ".*", java.util.Arrays.stream(parts).map(Pattern::quote).toArray(String[]::new));
```

Comment:
If we're building this on regex, this mean we're technically going to be misinterpreting fields like `http.status` (it'd match `http_status` too). We should escape existing dots in the pattern. Though the cases where this matters should be rare.